dts/bindings/flash_controller/zephyr,disk-flash.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Emulates a flash using a disk
+
+compatible: "zephyr,disk-flash"
+
+include: base.yaml
+
+properties:
+  size:
+    type: int
+    description: flash capacity in bits
+    required: true
+  disk-name:
+    type: string
+    description: the name of the underlying disk to be used
+    required: true
+  disk-sector-size:
+    type: int
+    description: the sector size in byte of the underlying disk to be used
+    required: true
+  disk-offset:
+    type: int
+    description: >
+      offset on the disk in bytes. If the offset is 10 the 10th byte will
+      be the first one used.
+    default: 0
+  init-level:
```

Comment:
I don't think this property or `init-priority` should be devicetree properties. Rather, we should use `POST_KERNEL` as the initialization level, and define a Kconfig `CONFIG_FLASH_DISK_INIT_PRIORITY` (if the init level needs to be different from `CONFIG_FLASH_INIT_PRIORITY`.)

Reply:
I thought that initialy too but then I discovered, that RAM-Disk is initialised in level "APPLICATION". As this driver is reliant on the fact that the underlying disk has already been initialized both `init-level` and `init-priority` need to be adjustable per instance as the underlying disk can be different depending on the instance.

Hovewer please do tell me if you have another idea how to handle this.

Reply:
> I thought that initialy too but then I discovered, that RAM-Disk is initialised in level "APPLICATION".

No, https://github.com/zephyrproject-rtos/zephyr/blob/e364f29bffdfb122fead4b7dad082428f2f6969c/drivers/disk/ramdisk.c#L169

Reply:
@jfischer-no Fair enough it is not anymore but at the time i wrote the driver it was ([ramdisk.c](https://github.com/zephyrproject-rtos/zephyr/blob/a5fd0d184a15d0af52599e0f5f5af880f353c6d4/drivers/disk/ramdisk.c)). Therefore initialization was not possible in post-kernel now it is.